geps/gep-2648/index.md

```diff
@@ -0,0 +1,415 @@
+reconciling the Policy type.
+
+#### On `Policy` objects
+
+Each Direct Attached Policy MUST populate the `Accepted` condition and reasons
+as defined in the Go spec, a snapshot of which is included below. The canonical
+representation is in the actual Go files. (At the time of writing, this is in
+`apis/v1alpha2/policy_types.go`)
+
+```go
+// PolicyConditionType is a type of condition for a policy.
+type PolicyConditionType string
+
+// PolicyConditionReason is a reason for a policy condition.
+type PolicyConditionReason string
+
+const (
+  // PolicyConditionAccepted indicates whether the policy has been accepted or rejected
+  // by a targeted resource, and why.
+  //
+  // Possible reasons for this condition to be True are:
+  //
+  // * "Accepted"
+  //
+  // Possible reasons for this condition to be False are:
+  //
+  // * "Conflicted"
+  // * "Invalid"
+  // * "TargetNotFound"
+  //
+  PolicyConditionAccepted PolicyConditionType = "Accepted"
+
+  // PolicyReasonAccepted is used with the "Accepted" condition when the policy has been
+  // accepted by the targeted resource.
+  PolicyReasonAccepted PolicyConditionReason = "Accepted"
+
+  // PolicyReasonConflicted is used with the "Accepted" condition when the policy has not
+  // been accepted by a targeted resource because there is another policy that targets the same
+  // resource and a merge is not possible.
+  PolicyReasonConflicted PolicyConditionReason = "Conflicted"
+
+  // PolicyReasonInvalid is used with the "Accepted" condition when the policy is syntactically
+  // or semantically invalid.
+  PolicyReasonInvalid PolicyConditionReason = "Invalid"
+
+  // PolicyReasonTargetNotFound is used with the "Accepted" condition when the policy is attached to
+  // an invalid target resource
+  PolicyReasonTargetNotFound PolicyConditionReason = "TargetNotFound"
+)
+```
+
+#### On targeted resources
+
+Implementations that use Direct Policy objects SHOULD put a Condition into
+`status.Conditions` of any objects affected by a Direct Policy, if that field
+is present.
+
+If they do, that Condition MUST have a `type` ending in `PolicyAffected` (like
+`gateway.networking.k8s.io/PolicyAffected`),
+and have the optional `observedGeneration` field kept up to date when the `spec`
+of the Policy-attached object changes.
+
+Implementations SHOULD use their own unique domain prefix for this Condition
+`type` - it is recommended that implementations use the same domain as in the
+`controllerName` field on GatewayClass (or some other implementation-unique
+domain for implementations that do not use GatewayClass).
+
+For objects that do _not_ have a `status.Conditions` field available (`Secret`
+is a good example), that object SHOULD instead have an annotation of
+`gateway.networking.k8s.io/PolicyAffected: true` (or with an
+implementation-specific domain prefix) added instead.
+
+Because these Conditions or annotations are namespaced per-implementation,
```

Comment:
Annotation or label?

Reply:
I've updated to label, although I originally chose annotation because these are machine-generated and machine-added. I updated because it seems useful to be able to search objects to find policy affected ones (which you can't do with annotations).